src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs

```diff
@@ -49,7 +49,7 @@ public partial class HubConnection
         };
 
         private static readonly MethodInfo _sendStreamItemsMethod = typeof(HubConnection).GetMethods(BindingFlags.NonPublic | BindingFlags.Instance).Single(m => m.Name.Equals(nameof(SendStreamItems)));
-#if NETCOREAPP3_0
+#if NETSTANDARD2_1
```

Comment:
Maybe all the `#if NETSTANDARD2_1` checks should be `#if !NETSTANDARD2_0`, that way we might never have to update them in the future.

Reply:
Doubt it matters. We can't be sure at this point when we'll drop .NET Standard 2.0 support, update 2.1 to something newer, add a new TFM, &hellip;

Reply:
What you can do is have ifdefs based on the available feature.

For example, this test could be `#if HAS_ASYNCENUMERABLE`. Then in the csproj against `NETSTANDARD2_1` you would define `HAS_ASYNCENUMERABLE`.

Now when a new target is added in the future, e.g. netcoreapp50, you just add it to the csproj and set the required flags. No need to update individual source code files.

Reply:
Example: https://github.com/JamesNK/Newtonsoft.Json.Bson/blob/a1db92678e7e72eb733f37079f3a93bfb6215338/Src/Newtonsoft.Json.Bson/Newtonsoft.Json.Bson.csproj#L36-L47

Reply:
We know that all targets after netstandard2.0 will have these new APIs so I don't see why we'd add extra upkeep when a simple `!NETSTANDARD2_0` check will do. And in the future if we remove netstandard2.0 compatibility then we just remove the `#if` entirely.

If for some crazy reason we do have future targets that remove features like `IAsyncEnumerable` then we can revisit and add feature flags.

Reply:
Future targets will never remove IAsyncEnumerable but future targets will add new features that you will likely want to use. With multiple targets things start getting complicated quickly...

A new feature in 3.0: `#if !(NETSTANDARD2_0 || NETSTANDARD2_1)`
A new feature in 4.0: `#if !(NETSTANDARD2_0 || NETSTANDARD2_1 || NETSTANDARD3_0)`
etc

Meanwhile using feature flags are a single test that and easy to read and all visible in the csproj to see what each target supports.